scraper/swimrankings.py

```
@@ -0,0 +1,205 @@
+
+        if a_name is None:
+            raise HTMLParsingError("Failed to find required html! Tag `a`")
+
+        last_name, first_name = a_name.get_text().split(', ')
+        last_name = last_name.title()
+
+        a_href = a_name['href']
+        try:
+            athlete_id = int(a_href[30:37])
+        except KeyError:
+            raise DataNotFoundError(f"Failed to find athlete_id!, Maybe html changed: `{a_name}`")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)